### PR TITLE
Fixes component form so that signal autocomplete input renders

### PR DESCRIPTION
**Pull request by johnclary**
## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/12810

Well, this one slipped through release testing :/ It stems from a minor change that Mike made at the end of the PR review process that I should have caught in testing. Ivonne brought it to my attention so I think we'll want to bring this forward as a patch release as well. 

## Testing
Local or Netlify (https://deploy-preview-1074--atd-moped-main.netlify.app/moped/projects)

**Steps to test:**
1. Create a new component. Select the  "Signal - Traffic" or "Signal - PHB" component
2. Observe that the autocomplete input appears and enables you to search/select a traffic signal or PHB.
3. Change the component type to something else. The autocomplete input goes away.
4. Save your signal component. Edit it. Observe that the autocomplete input is visible in the form.

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/auth/user.js [hunk 1]
```diff
@@ -109,7 +109,6 @@ export const initializeUserDBObject = (userObject) => {
     }).then((res) => {
       // Then we parse the response
       res.json().then((resData) => {
-        // If we have any user data,  use it
```
Comment (johnclary) [inline moped-editor/src/auth/user.js:112]: i'm v confused about the commit history in this PR. my local `git log` looks normal 🤷‍♂️ 

i will try rebasing after y'all review.

File: moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js [hunk 2]
```diff
@@ -92,7 +92,7 @@ const ComponentForm = ({
 
   const componentOptions = useComponentOptions(optionsData);
   const phaseOptions = usePhaseOptions(optionsData);
-  const { component, phase } = watch(["component", "phase"]);
+  const [component, phase] = watch(["component", "phase"]);
```
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js:95]: this was causing `component` to always be `undefined`, and so `isSignalComponent` always `false`.

File: moped-editor/src/auth/user.js (hunk 1 above)
Comment (mddilley) [reply moped-editor/src/auth/user.js:112]: definitely a puzzler! Ignored ✅

**Review by chiaberry — APPROVED** [review]
tested and the autocomplete is back!

**Review by mddilley — APPROVED** [review]
Womp womp 🎺 Thanks for fixing this as I could have caught it too (livin' on the edge 🏎️ 😅), and glad that Ivonne raised this up. 🙏 Test steps work ✅

File: moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js (hunk 2 above)
Comment (mddilley) [reply moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js:95]: I thought that this returning an array was missing from the docs, but the return is [clearly shown there](https://react-hook-form.com/docs/useform/watch). TIL!

Comment (johnclary) [reply moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js:95]: 👍

also the fact that using object destructuring on an array doesn't throw an error seems like a quintessential javascript bug 🤯

**Pull request closed without merging**